### PR TITLE
HHH-16563: Using jakarta.annotation.Generated instead of javax versions

### PR DESCRIPTION
**Pull request by TFyre**
https://hibernate.atlassian.net/browse/HHH-16563

### CODE REVIEW

**Comment by TFyre** [comment]
RFC: Should references to javax.annotation.Generated and javax.annotation.processing.Generated completely be removed?

**Comment by sebersole** [comment]
> RFC: Should references to javax.annotation.Generated and javax.annotation.processing.Generated completely be removed?

IMO yes.

**Comment by TFyre** [comment]
@sebersole, can you please review?

**Comment by sebersole** [comment]
One of 2 things needs to happen here:

1. Ideally, `hibernate-jpamodelgen` should be able to supply `jakarta.annotation:jakarta.annotation-api` as a transitive dependency into the `annotationProcessor` Configuration
2. The `jakarta.annotation:jakarta.annotation-api` dependencies you define here should not be added to `implementation`.  Worst case (aka, there is no way to achieve 1), those references should be changed to `testCompileOnly`.  There is no need for `jakarta.annotation:jakarta.annotation-api` to be on the runtime classpath for any Hibernate modules.

P.S. as I am unsure about the transitivity, I asked - https://discuss.gradle.org/t/transitive-dependencies-from-an-annotation-processor/45675

File: hibernate-micrometer/hibernate-micrometer.gradle [hunk 1]
```diff
@@ -2,13 +2,20 @@ description = 'Integration for Micrometer metrics into Hibernate as a metrics co
 
 apply from: rootProject.file( 'gradle/published-java-module.gradle' )
 
+configurations {
```
Comment (sebersole) [inline hibernate-micrometer/hibernate-micrometer.gradle:5]: These Configurations are transitive by default (see the Gradle forum post).  This is pointless.

Comment (TFyre) [reply hibernate-micrometer/hibernate-micrometer.gradle]: Whooops misunderstood 😄

File: hibernate-envers/hibernate-envers.gradle [hunk 2]
```diff
@@ -16,12 +16,14 @@ dependencies {
 
     implementation jakartaLibs.jaxbApi
     implementation jakartaLibs.jaxb
+    implementation jakartaLibs.annotation
```
[the remark this answers is not in the document]
Comment (TFyre) [reply hibernate-envers/hibernate-envers.gradle]: Whooops misunderstood 😄 t tests are failing

**Comment by sebersole** [comment]
https://github.com/gradle/gradle/issues/25187#issuecomment-1562892491

**Pull request merged**